api/v3/Contribution.php

```diff
@@ -472,28 +472,32 @@ function _civicrm_api3_contribution_sendconfirmation_spec(&$params) {
  */
 function civicrm_api3_contribution_completetransaction($params) {
   $input = $ids = [];
-  if (isset($params['payment_processor_id'])) {
-    $input['payment_processor_id'] = $params['payment_processor_id'];
-  }
+
   $contribution = new CRM_Contribute_BAO_Contribution();
   $contribution->id = $params['id'];
   if (!$contribution->find(TRUE)) {
     throw new API_Exception('A valid contribution ID is required', 'invalid_data');
   }
 
+  if (isset($params['payment_processor_id'])) {
+    $input['payment_processor_id'] = $params['payment_processor_id'];
+  }
```

Comment:
Moving this here prepares for a follow-up PR by making it clear that payment_processor_id is only used by loadRelatedObjects